File: GitToolBox/src/main/java/zielu/gittoolbox/extension/UpdateProjectActionEP.java
package zielu.gittoolbox.extension;

import com.intellij.openapi.extensions.ExtensionPointName;
import com.intellij.util.xmlb.annotations.Attribute;
import zielu.intellij.extensions.ZAbstractExtensionPointBean;

public class UpdateProjectActionEP extends ZAbstractExtensionPointBean {
  public static final ExtensionPointName<UpdateProjectActionEP>
      POINT_NAME = ExtensionPointName.create("zielu.gittoolbox.updateProjectAction");

  @Attribute("provider")
  public String provider;

  public UpdateProjectAction instantiate() {
    return createInstance(provider);
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/extension/update/UpdateProjectActionEP.kt
package zielu.gittoolbox.extension.update

import com.intellij.util.xmlb.annotations.Attribute
import zielu.intellij.extensions.ZAbstractExtensionPointBean

internal class UpdateProjectActionEP : ZAbstractExtensionPointBean() {
  @Attribute("provider")
  lateinit var provider: String

  fun instantiate(): UpdateProjectAction {
    return createInstance(provider)
  }
}
